src/main/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSink.java

```diff
@@ -258,4 +260,16 @@ private static void mergeInto(OutputStream out, Path directory, Configuration co
             fs.delete(part.getPath(), false);
         }
     }
+
+    protected static FileStatus[] getBamFragments( final Path directory, final FileSystem fs ) throws IOException {
+        final FileStatus[] parts = fs.globStatus(new Path(directory, "part-r-[0-9][0-9][0-9][0-9][0-9]*"));
+
+        // FileSystem.globStatus() has a known bug that causes it to not sort the array returned by
+        // name (despite claiming to): https://issues.apache.org/jira/browse/HADOOP-10798
+        // Because of this bug, we do an explicit sort here to avoid assembling the bam fragments
+        // in the wrong order.
+        Arrays.sort(parts);
```

Comment:
This workaround shall live forever as a defensive measure! :)

But, yes, we should update to Hadoop 2.8 when it's out.